src/gui/sdlmain.cpp

```diff
@@ -4870,15 +4876,16 @@ int sdl_main(int argc, char* argv[])
 #if defined(WIN32)
 		SetConsoleCtrlHandler((PHANDLER_ROUTINE)console_event_handler, TRUE);
 
-#if SDL_VERSION_ATLEAST(2, 24, 0)
-		if (SDL_SetHint(SDL_HINT_WINDOWS_DPI_AWARENESS, "permonitorv2") ==
-		    SDL_FALSE) {
-			LOG_WARNING("SDL: Failed to set DPI awareness flag");
-		}
-		if (SDL_SetHint(SDL_HINT_WINDOWS_DPI_SCALING, "1") == SDL_FALSE) {
-			LOG_WARNING("SDL: Failed to set DPI scaling flag");
+		if (is_runtime_sdl_version_at_least({2, 24, 0})) {
+			if (SDL_SetHint(SDL_HINT_WINDOWS_DPI_AWARENESS,
+			                "permonitorv2") == SDL_FALSE) {
+				LOG_WARNING("SDL: Failed to set DPI awareness flag");
+			}
+			if (SDL_SetHint(SDL_HINT_WINDOWS_DPI_SCALING, "1") ==
+			    SDL_FALSE) {
+				LOG_WARNING("SDL: Failed to set DPI scaling flag");
+			}
 		}
-#endif
 #endif // WIN32
```

Comment:
The intent of this one is different.  The macro `SDL_HINT_WINDOWS_DPI_AWARENESS` was added in 2.24 so the purpose of the compile time check is to allow compilation with older versions.  A runtime check doesn't make sense because presumably the function call will fail if the runtime SDL was too low and we would want to log that warning to the user.

These things are mostly here to support old versions of Debian Stable and such (whatever the oldest Linux distro we support is that may have an old SDL version).  As this check is inside an `#if defined(WIN32)` though, we can probably just remove the check all together.

Going forward (after this release) we may want to think about removing these once we get a fixed version solution working on Linux (vcpkg, Meson wrap, or equivalent).  These checks would really only benefit developers working on such distros.  The logic being if some distro is shipping Staging to end users and they aren't updating SDL, they probably aren't shipping the latest Staging either :slightly_smiling_face:

Reply:
Good point @weirddan455, TBH I was a bit unsure about this yesterday, but was also tired because of all the build talk, so this is what happened 😅 

I agree with your suggestions for the future; reverting this to be a compile-time check for now.